scripts/refresh_trends.py

```
@@ -145,33 +137,37 @@ def extract_trends(entries: list[dict]) -> dict | None:
 
-    for block in message.content:
-        if getattr(block, "type", None) == "tool_use" and block.name == TREND_TOOL["name"]:
-            return block.input  # type: ignore[return-value]
-    return None
+    text = (response.text or "").strip()
+    if not text:
+        return None
+    try:
+        parsed = json.loads(text)
+    except json.JSONDecodeError:
+        logger.error("Gemini returned non-JSON output: %r", text[:500])
```

Ruff (0.15.12)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)